### PR TITLE
More fixed_width ints and incorporating file_saving_test.c

### PR DESCRIPTION
**Pull request by hugbubby**
The file_saving_test.c was not included in the cmake list
and thus was ignored by travis and "make check". I found this
out while introducing ck_assert_msg into the integration test.

Also removed some variable width integers from encryptsave_test.c
and implemented ck_assert_msg, reorganized some loops,
and removed some longs in file_transfer_test.c.

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1025)

### CODE REVIEW

File: auto_tests/encryptsave_test.c [hunk 1]
```diff
@@ -207,12 +207,12 @@ static Suite *encryptsave_suite(void)
...
 {
     setvbuf(stdout, nullptr, _IONBF, 0);
-    srand((unsigned int) time(nullptr));
+    srand((uint32_t) time(nullptr));
 
     Suite *encryptsave =  encryptsave_suite();
     SRunner *test_runner = srunner_create(encryptsave);
 
-    int number_failed = 0;
+    uint8_t number_failed = 0;
```
Comment (iphydf) [inline auto_tests/encryptsave_test.c]: Maybe just get rid of the SRunner stuff. That's just old compat stuff. Now we just call the test function directly (and we shouldn't use START_TEST anymore, just declare a static function manually).

File: auto_tests/file_saving_test.c [hunk 2]
```diff
@@ -54,14 +55,11 @@ static void save_data_encrypted(void)
...
-                          &eerr)) {
-        fprintf(stderr, "error: could not encrypt, error code %d\n", eerr);
-        exit(4);
-    }
+    ck_assert_msg(tox_pass_encrypt(clear, size - TOX_PASS_ENCRYPTION_EXTRA_LENGTH, (const uint8_t *)pphrase, strlen(pphrase), cipher,
+                          &eerr), "Could not encrypt, error code %d.", eerr);
 
     size_t written_value = fwrite(cipher, sizeof(*cipher), size, f);
-    printf("written written_value = %u of %u\n", (unsigned)written_value, (unsigned)size);
+    printf("written written_value = %u of %u\n", (uint16_t)written_value, (uint16_t)size);
```
[the remark this answers is not in the document]
Comment (hugbubby) [reply auto_tests/file_saving_test.c]: woops

File: auto_tests/encryptsave_test.c (hunk 1 above)
Comment (hugbubby) [reply auto_tests/encryptsave_test.c]: I was never quite sure what SRunner does tbqh. Is my HEAD now equivalent?

File: auto_tests/encryptsave_test.c [hunk 3]
```diff
@@ -43,22 +43,21 @@ static void accept_friend_request(Tox *m, const uint8_t *public_key, const uint8
     }
 }
 
-START_TEST(test_known_kdf)
+static void test_known_kdf()
```
Comment (iphydf) [inline auto_tests/encryptsave_test.c]: test_known_kdf(void)

File: auto_tests/file_saving_test.c [hunk 4]
```diff
@@ -73,20 +72,18 @@ static void load_data_decrypted(void)
...
     fseek(f, 0, SEEK_END);
-    long size = ftell(f);
+    int64_t size = ftell(f);
     fseek(f, 0, SEEK_SET);
 
     uint8_t *cipher = (uint8_t *)malloc(size);
     uint8_t *clear = (uint8_t *)malloc(size - TOX_PASS_ENCRYPTION_EXTRA_LENGTH);
     size_t read_value = fread(cipher, sizeof(*cipher), size, f);
-    printf("read read_vavue = %u of %ld\n", (unsigned)read_value, size);
+    printf("Read read_vavue = %u of %ld\n", (uint8_t)read_value, size);
```
Comment (iphydf) [inline auto_tests/file_saving_test.c]: Undo cast.

File: auto_tests/file_transfer_test.c [hunk 5]
```diff
@@ -45,29 +45,20 @@ static uint64_t file_size;
...
 {
-    if (kind != TOX_FILE_KIND_DATA) {
-        ck_abort_msg("Bad kind");
-    }
+    ck_assert_msg(kind == TOX_FILE_KIND_DATA, "bad kind");
 
-    if (!(filename_length == sizeof("Gentoo.exe") && memcmp(filename, "Gentoo.exe", sizeof("Gentoo.exe")) == 0)) {
-        ck_abort_msg("Bad filename");
-    }
+    ck_assert_msg((filename_length == sizeof("Gentoo.exe")
```
Comment (iphydf) [inline auto_tests/file_transfer_test.c]: The outer () is now redundant.

File: auto_tests/file_transfer_test.c [hunk 6]
```diff
@@ -119,27 +105,19 @@ static bool file_sending_done;
...
 {
-    if (!sendf_ok) {
-        ck_abort_msg("Didn't get resume control");
-    }
+    ck_assert_msg(sendf_ok, "didn't get resume control");
 
-    if (sending_pos != position) {
-        ck_abort_msg("Bad position %llu", (unsigned long long)position);
-    }
+    ck_assert_msg(sending_pos == position, "bad position %llu", (unsigned long long) position);
```
Comment (iphydf) [inline auto_tests/file_transfer_test.c]: `%llu` is not supported on windows. Change to `%lu` and cast to `unsigned long`.

File: auto_tests/file_transfer_test.c [hunk 7]
```diff
@@ -149,16 +127,13 @@ static void tox_file_chunk_request(Tox *tox, uint32_t friend_number, uint32_t fi
...
     memset(f_data, sending_num, length);
 
-    if (tox_file_send_chunk(tox, friend_number, file_number, position, f_data, length, &error)) {
-        ++sending_num;
-        sending_pos += length;
-    } else {
-        ck_abort_msg("Could not send chunk, error num=%d pos=%d len=%d", (int)error, (int)position, (int)length);
-    }
+    ck_assert_msg(tox_file_send_chunk(tox, friend_number, file_number, position, f_data, length, &error),
+                  "Could not send chunk, error num=%d pos=%d len=%d", (int)error, (int)position, (int)length);
```
Comment (iphydf) [inline auto_tests/file_transfer_test.c]: Lowercase "could".

File: auto_tests/file_transfer_test.c [hunk 8]
```diff
@@ -149,16 +127,13 @@ static void tox_file_chunk_request(Tox *tox, uint32_t friend_number, uint32_t fi
...
 
-    if (error != TOX_ERR_FILE_SEND_CHUNK_OK) {
-        ck_abort_msg("Wrong error code");
-    }
+    ++sending_num;
+    sending_pos += length;
+
+    ck_assert_msg(error == TOX_ERR_FILE_SEND_CHUNK_OK, "Wrong error code");
```
Comment (iphydf) [inline auto_tests/file_transfer_test.c]: Lowercase "wrong".

File: auto_tests/file_transfer_test.c [hunk 9]
```diff
@@ -267,15 +238,13 @@ static void file_transfer_test(void)
...
-
-            ck_abort_msg("Something went wrong in file transfer %u %u %u %u %u %u %lu %lu %lu", sendf_ok, file_recv,
-                         totalf_size == file_size, size_recv == file_size, sending_pos == size_recv, file_accepted == 1,
-                         (unsigned long)totalf_size, (unsigned long)size_recv,
-                         (unsigned long)sending_pos);
+            ck_assert_msg(sendf_ok && file_recv && totalf_size == file_size && size_recv == file_size && sending_pos == size_recv
+                          && file_accepted == 1,
+                          "Something went wrong in file transfer %u %u %u %u %u %u %lu %lu %lu",
+                          sendf_ok, file_recv, totalf_size == file_size, size_recv == file_size, sending_pos == size_recv,
+                          file_accepted == 1, (unsigned long) totalf_size, (unsigned long) size_recv,
```
Comment (iphydf) [inline auto_tests/file_transfer_test.c]: No space after the cast (did you do this or did astyle?).

File: auto_tests/file_transfer_test.c [hunk 10]
```diff
@@ -297,7 +266,7 @@ static void file_transfer_test(void)
                   (unsigned long)totalf_size, (unsigned long)size_recv,
                   (unsigned long)sending_pos);
 
-    printf("100MiB file sent in %llu seconds\n", time(nullptr) - f_time);
+    printf("100MiB file sent in %lu seconds\n", time(nullptr) - f_time);
```
Comment (iphydf) [inline auto_tests/file_transfer_test.c]: Cast the result of the subtraction to `unsigned long`.

File: auto_tests/file_transfer_test.c [hunk 11]
```diff
@@ -327,31 +296,26 @@ static void file_transfer_test(void)
...
 
         c_sleep(min_u32(tox1_interval, min_u32(tox2_interval, tox3_interval)));
     }
 
-    printf("Starting file 0 transfer test.\n");
+    ck_assert_msg(sendf_ok && file_recv && m_send_reached && totalf_size == file_size && size_recv == max_sending
+                  && sending_pos == size_recv && file_accepted == 1,
+                  "something went wrong in file transfer %u %u %u %u %u %u %u %llu %llu %llu %llu", sendf_ok, file_recv,
+                  m_send_reached, totalf_size == file_size, size_recv == max_sending, sending_pos == size_recv, file_accepted == 1,
+                  (unsigned long long) totalf_size, (unsigned long long) file_size,
```
Comment (iphydf) [inline auto_tests/file_transfer_test.c]: No space after cast.

File: auto_tests/file_transfer_test.c [hunk 12]
```diff
@@ -413,3 +372,4 @@ int main(void)
     file_transfer_test();
     return 0;
 }
+
```
Comment (iphydf) [inline auto_tests/file_transfer_test.c]: Remove extra empty line.

**Comment by hugbubby** [comment]
dun

**Pull request merged**